tests/unit/python/foglamp/tasks/purge/test_purge.py

```diff
@@ -0,0 +1,148 @@
+              {"retainUnsent": {"value": "False"}, "age": {"value": "0"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "-1"}, "size": {"value": "-1"}}]
+
+    @pytest.mark.parametrize("conf, expected_return, expected_calls, valid_purge, valid_store_return", [
+        (config[0], (2, 4), {'sent_id': 0, 'size': '20', 'flag': 'purge'}, True, True),
+        (config[1], (1, 2), {'sent_id': 0, 'age': '72', 'flag': 'purge'}, True, True),
+        (config[2], (1, 2), {'sent_id': 0, 'size': '100', 'flag': 'purge'}, True, True),
+        (config[3], (2, 4), {'sent_id': 0, 'size': '20', 'flag': 'retain'}, True, True),
+        (config[4], (1, 2), {'sent_id': 0, 'age': '72', 'flag': 'retain'}, True, True),
+        (config[5], (1, 2), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, True, True),
+        (config[6], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, False, True),
+        (config[7], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, False, True),
+        (config[8], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, True, False)
+    ])
+    def test_purge_data(self, conf, expected_return, expected_calls, valid_purge, valid_store_return):
+        """Test that purge_data calls Storage's purge with defined values"""
+        with patch.object(FoglampProcess, '__init__'):
+            p = Purge()
+            p._logger = logger
+            p._logger.info = MagicMock()
+            p._logger.error = MagicMock()
+            p._storage = MagicMock(spec=StorageClient)
+            p._readings_storage = MagicMock(spec=ReadingsStorageClient)
+            audit = p._audit
+            with patch.object(p._readings_storage, 'purge', side_effect=self.store_purge) as mock_storage_purge:
+                with patch.object(audit, 'information', return_value=asyncio.ensure_future(asyncio.sleep(0.1))) \
+                        as audit_info:
+                    assert expected_return == p.purge_data(conf)
+                    # Test the positive case when all if conditions in purge_data pass
+                    if valid_purge and valid_store_return:
+                        assert audit_info.called
+                        args, kwargs = mock_storage_purge.call_args
+                        assert kwargs == expected_calls
+                    # Test the code block when no rows were purged
+                    elif not valid_purge and valid_store_return:
+                        p._logger.info.assert_called_once_with("No rows purged")
+                    # Test the code block when purge failed
+                    else:
+                        p._logger.error.assert_called_with('Purge failed: %s', '409 Conflict')
+            # Reset mocks required to clean any call states between parametrised tests
```

Comment:
we can avoid manual reset:

pytest-mock - installs a mocker fixture which is a thin-wrapper around the patching API provided by the mock package, but with the benefit of not having to worry about undoing patches at the end of a test.

Reply:
removed clean calls as we are using context managers, cleanup done when exiting from context managers.